lib/irmin.mli

```diff
@@ -2365,6 +2158,20 @@ module View (S: S): VIEW with type db = S.t
                           and type head = S.head
 (** Create views. *)
 
+val with_hrw_view :
+  (module S with type key = 'path) ->
+  (module VIEW with type t = 'view and type db = 'store and type key = 'path) ->
+  'store ->
```

Comment:
I think `S.t` should be constraint to be of type `'store` as well. Although it might be better to just remove that function and have a `View.with_transaction` function instead maybe? What do you think?